### PR TITLE
fix(preProcessPattern): throw an `{Error}` if `pattern.from` is empty or `undefined`

### LINKED ISSUE #278: The plugin should throw an error, if "from" is empty string
Hey guys. 
I spent hours trying to find a reason of weird behavior in my project, but it seems that `copy-webpack-plugin` takes the content of latest-processed file in case if **from** is an empty string, and put it inside the "to:" file. 
I think in such case the plugin should throw an error or show warning at least. 
As I checked in the plugin source, it is related to the wrong path resolving. 

Versions: 
* Webpack: 4.16.2
* copy-webpack-plugin: 4.5.2

Here is the example: 
**webpack.config.js**
```
const webpack = require('webpack');
const CopyWebpackPlugin = require('copy-webpack-plugin');

const config = {
	entry: './index.js',

	plugins: [
		new CopyWebpackPlugin([
			{
				from: '',
				to: './test.json'
			}
		]),
	]
};

module.exports = config;
```

**index.js**
```
console.log('Content from index.js');
```

Webpack run result: 
```
$ ./node_modules/.bin/webpack --mode=development
Hash: 73063a824e1b7c5d19e1
Version: webpack 4.16.2
Time: 1781ms
Built at: 2018-07-26 17:34:11
      Asset      Size  Chunks             Chunk Names
    main.js  3.78 KiB    main  [emitted]  main
./test.json  37 bytes          [emitted]  
Entrypoint main = main.js
[./index.js] 37 bytes {main} [built]
```

After the build content of **test.json** will be: 
```
console.log('Content from index.js');
```

### PR DESCRIPTION
**Pull request by Deliaz**
### Type

- [x] Fix

### Issues

- Fixes #278 

### SemVer

- [x] Patch

### CODE REVIEW

File: src/preProcessPattern.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ export default function preProcessPattern(globalRef, pattern) {
     pattern = typeof pattern === 'string' ? {
         from: pattern
     } : Object.assign({}, pattern);
+    if (pattern.from === '') {
```
Comment (michael-ciniawsky) [inline src/preProcessPattern.js]:
```js
if (!pattern.from) {
 throw new Error(
  'Copy Plugin\n\npattern.from must be defined\n\n(https://github.com/webpack-contrib/copy-webpack-plugin#from)\n'
 )
}

```

**Comment by Deliaz** [comment]
Just wondering, do I need to finish something? Not sure about the workflow here.

**Comment by Deliaz** [comment]
@michael-ciniawsky Hi, am I right that you need to sign CLA? :)

**Comment by evilebottnawi** [comment]
@Deliaz thanks

**Comment by Deliaz** [comment]
> @Deliaz thanks

Thanks. But I am still not sure, is there anything that is required from my side?

**Comment by evilebottnawi** [comment]
@Deliaz no, we can merge this

**Pull request merged**